mixnet/util/src/lib.rs

```diff
@@ -0,0 +1,42 @@
+use std::{collections::HashMap, net::SocketAddr, sync::Arc};
+
+use parking_lot::Mutex;
+use tokio::net::TcpStream;
+
+#[derive(Clone)]
+#[repr(transparent)]
+pub struct ConnectionPool {
+    pool: Arc<Mutex<HashMap<SocketAddr, Arc<tokio::sync::Mutex<TcpStream>>>>>,
+}
+
+impl ConnectionPool {
+    pub fn new(size: usize) -> Self {
```

Comment:
Sorry for the misleading. I will update the doc, because currently, we assume the mix node in each layer will not be too large. The size here just helps initialize the `HashMap` of the pool, and no eviction policy is applied.

Reply:
> Apart from the comments, I'm still having doubts about the solution as a whole.
Given that routes through the mixnet are supposed to be random or at least we can't expect specific patterns, keeping a subset of connections might not be very efficient (exact numbers depend on subset size ofc).
If we can guarantee the pool to be as big as necessary (e.g. the next layer) then this is good, but otherwise I would like to see measurements to see whether this helps or not and then some documentation on our findings.
Actually, I do not see any limit on the size of the connection pool, so it grows as big as necessary atm, is that your intention or you forgot to add it (there's some mention of it in the doc but that parameter does not actually do anything)?

I think we can discuss about this topic here. 

Yeah, since the current implementation is the simplest approach, we don't have eviction. Though, I thought about boostrapping all connections with the next mix layer at the beginning. Instead, now we do lazy-init. They have pros and cons, but lazy-init might be easier to implement for now. 
For now, IMO, it's enough to set the max size of hashmap to the size of the next mix layer. If it hits the max, we could panic or return an error. That won't happen unless the sender picked the wrong route.

Reply:
We can simply run this test: https://github.com/logos-co/nomos-node/pull/326/files#diff-de86edb550dad603a64ce6e7b47a7e62f68a2b91ddc8c6e96dd858cb4a47c69bR55, to see if this PR solves the `ran out of fds` error. Plus, we need to check if this PR improves the overall latency.

Reply:
if this solution is a lazy-init then it's fine (although we have some 2 locks to access a connection).
Also atm we don't have any limit on the max size of the hashmap, but also I do not seed a need for it